pkg/controller/migcluster/migcluster_controller.go

```diff
@@ -76,17 +89,76 @@ type ReconcileMigCluster struct {
 // +kubebuilder:rbac:groups=migration.openshift.io,resources=migclusters,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=migration.openshift.io,resources=migclusters/status,verbs=get;update;patch
 func (r *ReconcileMigCluster) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Info(fmt.Sprintf("[mCluster] RECONCILE [nsName=%s/%s]", request.Namespace, request.Name))
+
 	// Fetch the MigCluster instance
 	instance := &migrationv1alpha1.MigCluster{}
 	err := r.Get(context.TODO(), request.NamespacedName, instance)
 	if err != nil {
 		if errors.IsNotFound(err) {
-			// Object not found, return.  Created objects are automatically garbage collected.
-			// For additional cleanup logic use finalizers.
-			return reconcile.Result{}, nil
+			return reconcile.Result{}, nil // don't requeue
+		}
+		return reconcile.Result{}, err // requeue
+	}
+
+	// Get the host cluster object
+	isHostCluster := instance.Spec.IsHostCluster
+	log.Info(fmt.Sprintf("[mCluster] isHostCluster: [%v]", isHostCluster))
+
+	// Get the SA secret attached to MigCluster
+	saSecretRef := instance.Spec.ServiceAccountSecretRef
+	saSecret := &kapi.Secret{}
+	err = r.Get(context.TODO(), types.NamespacedName{Name: saSecretRef.Name, Namespace: saSecretRef.Namespace}, saSecret)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return reconcile.Result{}, nil // don't requeue
 		}
-		// Error reading the object - requeue the request.
-		return reconcile.Result{}, err
+		return reconcile.Result{}, err // requeue
+	}
+	saTokenKey := "saToken"
+	saTokenData, ok := saSecret.Data[saTokenKey]
+	if !ok {
+		log.Info(fmt.Sprintf("[mCluster] saToken: [%v]", ok))
+		return reconcile.Result{}, nil // don't requeue
 	}
+	saToken := string(saTokenData)
+	// log.Info(fmt.Sprintf("saToken: [%s]", saToken))
+
+	// Get k8s URL from cluster-registry Cluster associated with MigCluster
+	crClusterRef := instance.Spec.ClusterRef
```

Comment:
If `instance` was renamed to `migCluster`, it would be clearer for `crClusterRef` should just be `ClusterRef` since the _kind_ is `Cluster`.